### PR TITLE
Fix Escrow Repay

### PR DESCRIPTION
**Pull request by passabilities**
It:
* Sets the allowance on the Lending Pool instead of the Loans contract
* Only takes the necessary amount of funds to repay the loan if Escrow balance does not cover the amount

### CODE REVIEW

File: contracts/base/Escrow.sol [hunk 1]
```diff
@@ -170,12 +170,17 @@ contract Escrow is EscrowInterface, TInitializable, BaseUpgradeable, BaseEscrowD
...
         uint256 balance = _balanceOf(loans.lendingToken());
         if (amount > balance) {
+            uint256 totalOwed = loans.getTotalOwed(loanID);
+            uint256 amountNeeded = amount > totalOwed
+                ? totalOwed.sub(balance)
+                : amount.sub(balance);
+
             require(
-                token.transferFrom(msg.sender, address(this), amount.sub(balance)),
+                token.transferFrom(msg.sender, address(this), amountNeeded),
```
Comment (salazarguille) [inline contracts/base/Escrow.sol:179]: Do we need to validate the returned value?

File: contracts/base/Escrow.sol [hunk 2]
```diff
@@ -170,12 +170,17 @@ contract Escrow is EscrowInterface, TInitializable, BaseUpgradeable, BaseEscrowD
         IERC20 token = IERC20(loans.lendingToken());
         uint256 balance = _balanceOf(loans.lendingToken());
         if (amount > balance) {
+            uint256 totalOwed = loans.getTotalOwed(loanID);
+            uint256 amountNeeded = amount > totalOwed
```
Comment (salazarguille) [inline contracts/base/Escrow.sol:174]: I think there is a potential underflow. 
Example:
amount: 60
totalOwed: 30
balance: 40

Then the amount > balance and amount > totalOwed are true, and the substraction ```totalOwed.sub(balance)``` will fail.

**Review by salazarguille — CHANGES_REQUESTED** [review]

**Review by salazarguille — APPROVED** [review]
LGTM

File: contracts/base/Escrow.sol [hunk 3]
```diff
@@ -169,13 +169,18 @@ contract Escrow is EscrowInterface, TInitializable, BaseUpgradeable, BaseEscrowD
     function repay(uint256 amount) external onlyOwner {
         IERC20 token = IERC20(loans.lendingToken());
```
Comment (Juan-cc) [inline contracts/base/Escrow.sol:170]: Suggestion: I would rename token to lendingToken to refer that specific token, and replace other occurrences.

File: contracts/base/Escrow.sol [hunk 4]
```diff
@@ -169,13 +169,18 @@ contract Escrow is EscrowInterface, TInitializable, BaseUpgradeable, BaseEscrowD
...
         uint256 balance = _balanceOf(loans.lendingToken());
```
Comment (Juan-cc) [inline contracts/base/Escrow.sol:171]: Replace loans.lendingToken() with token (or lendingToken if previous suggestion is applied)

**Pull request merged**